eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -592,10 +600,19 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
       context.system.scheduler.scheduleOnce(2 seconds, self, remoteAnnSigs)
       stay
 
-    case Event(getTxResponse: GetTxWithMetaResponse, d: DATA_WAIT_FOR_FUNDING_CONFIRMED) if getTxResponse.txid == d.commitments.commitInput.outPoint.txid => handleGetFundingTx(getTxResponse, d.waitingSince, d.fundingTx)
+    case Event(getTxResponse: GetTxWithMetaResponse, d: DATA_WAIT_FOR_FUNDING_CONFIRMED) if getTxResponse.txid == d.commitments.commitInput.outPoint.txid => handleGetFundingTx(getTxResponse, d.waitingSinceBlock, d.fundingTx)
 
     case Event(BITCOIN_FUNDING_PUBLISH_FAILED, d: DATA_WAIT_FOR_FUNDING_CONFIRMED) => handleFundingPublishFailed(d)
 
+    case Event(c: CurrentBlockCount, d: DATA_WAIT_FOR_FUNDING_CONFIRMED) => d.fundingTx match {
+      case Some(_) => stay // we are funder, we're still waiting for the funding tx to be confirmed
+      case None if c.blockCount - d.waitingSinceBlock > FUNDING_TIMEOUT_FUNDEE =>
+        log.warning(s"funding tx hasn't been published in ${c.blockCount - d.waitingSinceBlock} blocks")
+        self ! BITCOIN_FUNDING_TIMEOUT
+        stay
+      case None => stay // let's wait longer
+    }
+
```

Comment:
This needs to be done in `OFFLINE`/`SYNCING` state too I believe.

Edit: saw your comment
> There is one edge case where a channel in WAIT_FOR_FUNDING_CONFIRMED will not be automatically cleaned up: if the peer never connects back to you, the channel will stay in OFFLINE and won't check for funding timeout. I think this edge case is quite harmless and can be ignored (I don't think it's worth adding more code to fix it).

I can definitely think of a scenario when peers open channels to us, then fail to publish and disappear. What makes it okay is that we will clean up at next startup.

Reply:
Exactly, I think this is ok and not worth adding more code.